### ISSUE TITLE
Were the models of Dyer.2022, Grazzini 2017, and Platt 2021 included in our black-it?

### ISSUE THREAD

**Issue by linwenqi2024** [bancaditalia/black-it#111]
Dyer 2022: Black-box Bayesian inference for agent-based models, Platt 2021: Bayesian Estimation of Economic Simulation Models Using Neural Networks, Grazzini 2017 Bayesian estimation of agent-based models, were the latest advances in the field. It would be appreciated if they were included, and promote largely the value of black-it.

**Comment by aldoglielmo** [comment]
Hi @linwenqi2024, thank you for opening this issue. Black-it should have already most of the search methods and loss functions used within the Platt 2021 paper as well as Grazzini 2017, have a look at the [cors samples](https://github.com/bancaditalia/black-it/blob/main/black_it/samplers/cors.py) , [particle swarm sampler](https://github.com/bancaditalia/black-it/blob/main/black_it/samplers/particle_swarm.py) , [halton sampler](https://github.com/bancaditalia/black-it/blob/main/black_it/samplers/halton.py)  and [gsl loss](https://github.com/bancaditalia/black-it/blob/main/black_it/loss_functions/gsl_div.py) , [likelihood loss](https://github.com/bancaditalia/black-it/blob/main/black_it/loss_functions/likelihood.py).

We do not currently have the Dyer 2022 methods. To incorporate the methods in that paper, the easiest option would probably be to build an interface to the sbi package https://github.com/sbi-dev/sbi.  I can help you out if you give it a try!

**Comment by linwenqi2024** [comment]
Thanks a lot. if possible, I would like to take a try if you could guide me in details and step by step. I'm a new comer in this field, and it may take a long time to get the results.

**Comment by aldoglielmo** [comment]
Hi @linwenqi2024, 
Well, this is certainly a project where you can learn a lot. If I were you, I would first familiarise myself with both Black-it and the SBI package by running a few tutorial examples from both. Once you understand the basics, you also need to learn the inner workings. Black-it has very intuitive and transparent interfaces (essentially the "Calibrator" object calls alternatively a "[Sampler](https://github.com/bancaditalia/black-it/blob/main/black_it/samplers/base.py)"  and a "[Loss function](https://github.com/bancaditalia/black-it/blob/main/black_it/loss_functions/base.py)", as also described [here](https://bancaditalia.github.io/black-it/description/)), but I am not an expert of SBI, and you'll need to understand how that works too. 

At a very abstract level, it should definitely be possible to take the building blocks of SBI and conform (some of) them to the "Sampler" and "Loss function" interfaces I just mentioned, so that one can instantiate and call them directly within the Black-it framework
